cuqi/sampler/_conjugate.py

```diff
@@ -0,0 +1,37 @@
+from cuqi.distribution import Posterior, GaussianCov, GaussianPrec, Gamma, GMRF
+import numpy as np
+
+class Conjugate: # TODO: Subclass from Sampler once updated
+    """ Conjugate sampler
+
+    Sampler for sampling a posterior distribution where the likelihood and prior are conjugate.
+
+    Currently supported conjugate pairs are:
+    - (GaussianPrec, Gamma)
+    - (GaussianCov, Gamma)
+    - (GMRF, Gamma)
+
+    For more information on conjugate pairs, see https://en.wikipedia.org/wiki/Conjugate_prior.
+
+    """
+
+    def __init__(self, target: Posterior):
+        if not isinstance(target.likelihood.distribution, (GaussianPrec, GaussianCov, GMRF)):
+            raise ValueError("Conjugate sampler only works with GaussianPrec likelihood function")
```

Comment:
Is it intentional that the checks looks for `(GaussianPrec, GaussianCov, GMRF)` while the error message mention `GaussianPrec` only.

Reply:
Good catch. Updated text